### PR TITLE
Add --diff support for ldap_attrs module

### PR DESCRIPTION
**Pull request by StopMotionCuber**
##### SUMMARY

This adds support for the  `--diff` mode for the `ldap_attrs` module. For the other `ldap` modules this does not really make sense, as `ldap_search` is read-only and for `ldap_entry` there is only a change if the entry is not there at all (so the state is quite clear).
We would like to use this kind of feature for internal use of the module.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`ldap_attrs`

##### ADDITIONAL INFORMATION

Right now, the module does not search for all available attributes at the same time. The `--diff` mode therefore has difficulties if we are only adding/deleting single values of a multi value attribute to detect what was there before and what would be the final state.
With `--exact`, we're replacing everything, therefore this issue is not present there. 
I'm not sure about the policy on this repo, whether a potentially misleading  diff is worse than none. Could also change that to not use "before" and "after", but going for "prepared" and a message like "`<key>: Added value(s) [<value1>, <value2>]" for the `present` or `absent` state. We are only using the `exact` state, so partial diff support only for that state would be fine to us.

Furthermore, attributes are not shown in the `--diff` if they are on the server, but not used within the module (probably not an issue).

Yeah well, as you can see from the longish message here I also want to get the discussion going. I'll add the fragment as soon as the discussions here are cleared up and the path for the PR is more clear.

### CODE REVIEW

File: plugins/modules/ldap_attrs.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
   check_mode:
     support: full
   diff_mode:
-    support: none
+    support: full
```
Comment (felixfontein) [inline plugins/modules/ldap_attrs.py:47]:
```suggestion
    support: full
    version_added: 8.5.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: plugins/modules/ldap_attrs.py [hunk 2]
```diff
@@ -309,16 +326,18 @@ def main():
 
     # Instantiate the LdapAttr object
     ldap = LdapAttrs(module)
+    old = None
+    new = None
```
Comment (russoz) [inline plugins/modules/ldap_attrs.py:330]: The nit picker in me would like to suggest a different name as `new` is a Python keyword (in the context of classes), just for the sake of differentiating from that.

Comment (felixfontein) [reply plugins/modules/ldap_attrs.py:330]: I don't think `new` is a keyword in Python. The language reference does not list it as a keyword: https://docs.python.org/3/reference/lexical_analysis.html#keywords

Comment (StopMotionCuber) [reply plugins/modules/ldap_attrs.py]: In python it's not a keyword. When switching back and forth between languages that might still be confusing though, therefore I changed all occurences to `[old|new]_attrs`

**Comment by StopMotionCuber** [comment]
> Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

Have added the changelog fragment and added `version_added` for diff mode.
Thank you for your quick response and review of the MR

File: plugins/modules/ldap_attrs.py (hunk 2 above)
Comment (russoz) [reply plugins/modules/ldap_attrs.py]: Again my jumping the gun. I was thinking of the `__new__` method in classes. Anyways, I am happy that the vars now have more meaning in their names.

File: changelogs/fragments/8073-ldap-attrs-diff.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - ldap_attrs - Module now supports diff mode, showing which attributes are changed within an operation
```
Comment (russoz) [inline changelogs/fragments/8073-ldap-attrs-diff.yml:2]: Please check the documentation and other fragment files as examples. We are missing the link to this PR (and the issue url if there was one).

File: plugins/modules/ldap_attrs.py (hunk 1 above)
Comment (russoz) [reply plugins/modules/ldap_attrs.py:47]: @felixfontein 
Isn't this something that could/should be replace by the doc fragment ?

**Review by russoz — COMMENTED** [review]
Code-wise it looks good, now just a couple of minor adjustments for the documentation.

File: plugins/modules/ldap_attrs.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/ldap_attrs.py:47]: No, `support` needs to be specified per module, and `version_added` needs to be added if `support` in increased after the module was added. This is nothing the doc fragment can really take over (except for special classes of modules, like `_facts` and `_info` modules, where `support` must have a specific value).

File: changelogs/fragments/8073-ldap-attrs-diff.yml (hunk 3 above)
Comment (StopMotionCuber) [reply changelogs/fragments/8073-ldap-attrs-diff.yml]: Yup, has been adapted. Only skimmed thorough the docs, so I missed that there should be a reference to the PR.
There was no issue to this, felt like that would be pointless if it's just for the sake creating a PR right after.

**Review by russoz — APPROVED** [review]
LGTM

**Review by felixfontein — COMMENTED** [review]
One nit for the changelog fragment :)

File: changelogs/fragments/8073-ldap-attrs-diff.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - ldap_attrs - module now supports diff mode, showing which attributes are changed within an operation (https://github.com/ansible-collections/community.general/pull/8073)
```
[the remark this answers is not in the document]
Comment (StopMotionCuber) [reply changelogs/fragments/8073-ldap-attrs-diff.yml]: Fine, I'll let you have that byte 😄

**Comment by felixfontein** [comment]
@StopMotionCuber thanks for your contribution!
@russoz thanks for reviewing!

**Pull request merged**